src/Conversion/ONNXToKrnl/Tensor/Tile.cpp

```diff
@@ -58,84 +59,62 @@ struct ONNXTileOpLowering : public ConversionPattern {
     ONNXTileOp tileOp = llvm::cast<ONNXTileOp>(op);
     auto loc = op->getLoc();
 
-    // get input operands, shapes, and rank
-    Value input = operandAdaptor.input();
-    auto inputMemRefType = input.getType().cast<MemRefType>();
-    auto inputShape = inputMemRefType.getShape();
-    int64_t inputRank = inputShape.size();
-    Value repeats = operandAdaptor.repeats();
+    ONNXTileOpShapeHelper shapeHelper(&tileOp, &rewriter);
+
+    if (failed(shapeHelper.Compute(operandAdaptor)))
```

Comment:
nit: since it already succeeded in shape inference, you could just do an assert here

assert(succeeded(shapeHelper.Compute(operandAdaptor)) && "expected success");